src/Symfony/Component/Workflow/Dumper/GraphvizDumper.php

```diff
@@ -84,7 +84,7 @@ private function findTransitions(Definition $definition)
         $transitions = array();
 
         foreach ($definition->getTransitions() as $transition) {
-            $transitions[] = array(
+            $transitions[$transition->getName()] = array(
```

Comment:
>If they have the same name, they have different from/to, so they should both be rendered on the graph separately (otherwise, the graph will lie).

That not true for the way state machines have been implemented. It splits the transition that have many tos in many transition with the same name.

See https://github.com/symfony/symfony/blob/master/src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php#L416.